client/my-sites/theme/main.jsx

```diff
@@ -1026,6 +1055,43 @@ class ThemeSheet extends Component {
 		);
 	};
 
+	getPremiumGlobalStylesEventProps = () => {
+		const { selectedStyleVariationSlug, themeId } = this.props;
+		return {
+			theme_name: themeId,
+			style_variation: selectedStyleVariationSlug ?? DEFAULT_VARIATION_SLUG,
+		};
+	};
+
+	onPremiumGlobalStylesUpgradeModalCheckout = () => {
+		this.props.recordTracksEvent(
+			'calypso_theme_sheet_global_styles_gating_modal_checkout_button_click',
+			this.getPremiumGlobalStylesEventProps()
+		);
+
+		this.setState( { showUnlockStyleUpgradeModal: false } );
+		page( `/checkout/${ this.props.siteSlug || '' }/premium` );
```

Comment:
We have to set the `redirect_to` query string so that they're able to come back to this page when they finish the checkout. Otherwise, they will see the thank you page

Reply:
Good catch! I've set the `redirect_to` param 🙂